chains/orchestrator-relays/runtime/dancelight/src/tests/ethereum_token_transfers.rs

```diff
@@ -1338,6 +1341,267 @@ fn test_pricing_parameters() {
+                "00000000000000000000006e61746976655f746f6b656e5f7472616e73666572"
+            ));
+            let agent_id = AgentId::from_low_u64_be(10);
+            let para_id: ParaId = 2000u32.into();
+            let amount_to_transfer = 10_000u128;
+
+            assert_ok!(EthereumTokenTransfers::set_token_transfer_channel(
+                root_origin(),
+                channel_id,
+                agent_id,
+                para_id
+            ));
+
+            // Define a mock ERC20 token address
+            let token_address = H160(hex!("deadbeefdeadbeefdeadbeefdeadbeefdeadbeef"));
+
+            let erc20_asset_location = Location {
+                parents: 1,
+                interior: X2([
+                    GlobalConsensus(EthereumNetwork::get()),
+                    AccountKey20 {
+                        network: Some(EthereumNetwork::get()),
+                        key: token_address.into(),
+                    },
+                ]
+                .into()),
+            };
+
+            let asset_id = 42u16;
+
+            assert_ok!(ForeignAssetsCreator::create_foreign_asset(
+                root_origin(),
+                erc20_asset_location.clone(), // Use the ERC20 location
+                asset_id,
+                AccountId::from(ALICE),
+                true,
+                1
+            ));
+
+            // Give tokens to user as if tokens were bridged before + extra to check the correct
+            // amount is sent
+            ForeignAssets::mint_into(asset_id, &AccountId::from(BOB), amount_to_transfer + 10)
+                .expect("to mint amount");
+
+            // User tries to send tokens
+            let beneficiary_address = H160(hex!("0123456789abcdef0123456789abcdef01234567"));
+
+            // Beneficiary location must be represented using destination's point of view.
+            let beneficiary_location = Location {
+                parents: 0,
+                interior: X1([AccountKey20 {
+                    network: Some(EthereumNetwork::get()),
+                    key: beneficiary_address.into(),
+                }]
+                .into()),
+            };
+
+            let eth_asset = AssetId(erc20_asset_location.clone())
+                .into_asset(Fungibility::Fungible(amount_to_transfer));
+
+            let balance_before = Balances::balance(&AccountId::from(BOB));
+
+            assert_ok!(XcmPallet::transfer_assets(
+                RuntimeOrigin::signed(AccountId::from(BOB)),
+                Box::new(EthereumLocation::get().into()),
+                Box::new(beneficiary_location.into()),
+                Box::new(vec![eth_asset].into()),
+                0u32,
+                Unlimited,
+            ));
+
+            // Correct amount has been sent
+            assert_eq!(ForeignAssets::balance(asset_id, &AccountId::from(BOB)), 10);
+
+            // Check fees have been payed (TODO: expect specific amount?)
```

Comment:
Do not forget to either update the comment or specify the fee amount in the test.

Reply:
Currently I just check the balance of native tokens decreased and thus some fees have been payed. I currently don't know how to get the expected fee.